### ISSUE TITLE
"Error: Failed to obtain access token" when opening a new project (installed via docker-compose.yml with HTTPS)

### ISSUE THREAD

**Issue by dvdjaco** [flowforge/flowforge#1854]
### Current Behavior

I deploy on a clean Debian 11 system using the Docker method [as described in the docuemntation](https://flowforge.com/docs/install/docker/). The default install with HTTP works fine, but when I follow the steps to use HTTPS I see the following issue:

When I create a new Node-RED instance on FlowForge and open the "Editor" I get the error "Failed to obtain access token":
![Screenshot from 2023-03-17 09-50-37](https://user-images.githubusercontent.com/1502076/225862049-c90b873c-d39b-493f-92ba-9d8d624407fc.png)

The logs from the `forge` container show an unknown oauth error (replaced the real domain with my.domain):
```
# docker logs flowforge-forge-1
[2023-03-17T08:24:33.842Z] INFO: FlowForge v1.5.0
[2023-03-17T08:24:33.843Z] INFO: FlowForge running with NodeJS v16.19.1
[2023-03-17T08:24:33.843Z] INFO: FlowForge Data Directory: /usr/src/forge
[2023-03-17T08:24:33.843Z] INFO: Config File: /usr/src/forge/etc/flowforge.yml
[2023-03-17T08:24:33.928Z] INFO: Database driver: postgres
[2023-03-17T08:24:35.033Z] INFO: No license applied
[2023-03-17T08:24:35.039Z] INFO: Usage       : count/limit
[2023-03-17T08:24:35.040Z] INFO:  Users      : 1/150
[2023-03-17T08:24:35.040Z] INFO:  Teams      : 1/50
[2023-03-17T08:24:35.040Z] INFO:  Projects   : 1/50
[2023-03-17T08:24:35.040Z] INFO:  Devices    : 0/50
[2023-03-17T08:24:35.903Z] INFO: Email not configured
[2023-03-17T08:24:36.137Z] INFO: Container driver: docker
[2023-03-17T08:24:36.261Z] INFO: Server listening at http://0.0.0.0:3000
[2023-03-17T08:24:36.261Z] INFO: ****************************************************
[2023-03-17T08:24:36.261Z] INFO: * FlowForge is now running and can be accessed at: *
[2023-03-17T08:24:36.261Z] INFO: *   http://forge.nodered.my.domain                 *
[2023-03-17T08:24:36.262Z] INFO: ****************************************************
[2023-03-17T08:24:36.408Z] INFO: Connected to comms broker
[2023-03-17T08:25:24.290Z] ERROR: AUDIT: event: auth.login.fail.oauth, type: project, id: 2bd680ff-719f-45b8-9956-80cf379761fd, unknown error
[2023-03-17T08:49:46.991Z] ERROR: AUDIT: event: auth.login.fail.oauth, type: project, id: 2bd680ff-719f-45b8-9956-80cf379761fd, unknown error
```
These are the "Node-RED" logs as shown in the FlowForge UI:
```
3/16/2023 6:27:31 PM  [system]  Launcher Started
3/16/2023 6:27:32 PM  [system]  Loading project settings
3/16/2023 6:27:32 PM  [system]  Target state is 'running'
3/16/2023 6:27:32 PM  [system]  Starting Node-RED
3/16/2023 6:27:43 PM  [info]  Welcome to Node-RED
===================
3/16/2023 6:27:43 PM  [info]  Node-RED version: v3.0.2
3/16/2023 6:27:43 PM  [info]  Node.js  version: v16.16.0
3/16/2023 6:27:43 PM  [info]  Linux 5.10.0-21-cloud-amd64 x64 LE
3/16/2023 6:27:51 PM  [info]  Loading palette nodes
3/16/2023 6:27:53 PM  [info]  FlowForge HTTP Authentication Plugin loaded
3/16/2023 6:27:53 PM  [info]  FlowForge Team Library Plugin loaded
3/16/2023 6:27:53 PM  []  FlowForge Light Theme Plugin loaded
3/16/2023 6:27:53 PM  []  FlowForge Dark Theme Plugin loaded
3/16/2023 6:28:04 PM  [warn]  ------------------------------------------------------
3/16/2023 6:28:04 PM  [warn]  [@flowforge/nr-project-nodes/project-link] Error: Project Link nodes cannot be loaded outside of FlowForge EE environment (line:6)
3/16/2023 6:28:04 PM  [warn]  [@flowforge/nr-file-nodes/file] 'file in' already registered by module node-red
3/16/2023 6:28:04 PM  [warn]  ------------------------------------------------------
3/16/2023 6:28:04 PM  [info]  Settings file  : /data/settings.js
3/16/2023 6:28:04 PM  [info]  Context store  : 'default' [module=memory]
3/16/2023 6:28:04 PM  [info]  Server now running at http://127.0.0.1:1880/
3/16/2023 6:28:04 PM  [warn]  Encrypted credentials not found
3/16/2023 6:28:04 PM  [info]  Starting flows
3/16/2023 6:28:04 PM  [info]  Started flows
```

### Expected Behavior

The OAuth login should work on a new install when following the documentation. 

### Steps To Reproduce

- Install using the Docker configuration provided in the official docs.
- Configure HTTPS following the documentation.
- Start FlowForge with `docker compose -p flowforge up -d`
- Do the initial setup.
- Create a new project.
- Start the Node-RED instance and open the Node-RED editor.

### Environment

- Forge docker image: flowforge/forge-docker:latest (sha 17d49591, created on 2023-03-16T11:04:29.303644007Z)
  - FlowForge version: v1.5.0
  - Node.js version: v16.19.1
  - npm version: 8.19.3
- Platform/OS: Debian GNU/Linux 11, Docker version 23.0.1
- Browser: Firefox 112

My docker-compose.yml (replaced the real domain with `my.domain`):
```
version: "3.3"
services:
  nginx:
    image: nginxproxy/nginx-proxy
    networks:
      - flowforge 
    restart: always
    volumes:
      - "/var/run/docker.sock:/tmp/docker.sock:ro"
      - "./nginx/vhost.d:/etc/nginx/vhost.d"
      - "./nginx/html:/usr/share/nginx/html"
      - "./certs:/etc/nginx/certs"
    ports:
      - "80:80"
      - "443:443"
    environment:
      - "HTTPS_METHOD=redirect"
  acme:
     image: nginxproxy/acme-companion
     volumes:
       - "/var/run/docker.sock:/var/run/docker.sock:ro"
       - "./acme:/etc/acme.sh"
     volumes_from:
       - nginx:rw
     environment:
       - "DEFAULT_EMAIL=mail@nodered.my.domain"
     depends_on:
       - "nginx"
  postgres:
    image: postgres:14
    networks: 
      - flowforge
    restart: always
    environment:
      POSTGRES_PASSWORD: secret
      POSTGRES_USER: root
    volumes:
      - "./db:/var/lib/postgresql/data"
      - "./setup-db.sh:/docker-entrypoint-initdb.d/01-setup-db.sh"
      - "./setup-context-db.sh:/docker-entrypoint-initdb.d/02-setup-context-db.sh"
  flowforge-broker:
    image: "iegomez/mosquitto-go-auth"
    networks:
      - flowforge
    restart: always
    ulimits:
      nofile: 2048
    environment:
      - "VIRTUAL_HOST=mqtt.nodered.my.domain"
      - "VIRTUAL_PORT=1884"
      - "LETSENCRYPT_HOST=mqtt.nodered.my.domain"
    volumes:
      - "./broker/mosquitto.conf:/etc/mosquitto/mosquitto.conf"
  forge:
    build:
      context: "./flowforge-docker"
    image: "flowforge/forge-docker"
    networks: 
      - flowforge
    restart: always
    environment:
      - "VIRTUAL_HOST=forge.nodered.my.domain"
      - "LETSENCRYPT_HOST=forge.nodered.my.domain"
    volumes:
      - "/var/run/docker.sock:/tmp/docker.sock"
      - "./etc/flowforge.yml:/usr/src/forge/etc/flowforge.yml"
    depends_on:
      - "postgres"
      - "nginx"
      - "flowforge-broker"

  file-server:
    build:
      context: "./file-server"
    image: "flowforge/file-server"
    networks:
      - flowforge
    restart: always
    volumes:
      - "./etc/flowforge-storage.yml:/usr/src/flowforge-file-server/etc/flowforge-storage.yml"
      - "./fileStorage:/usr/src/flowforge-file-server/var/root"
    depends_on:
      - "forge"

networks:
  flowforge:
```

My `etc/flowforge.yml` file:
```
port: 3000
host: 0.0.0.0
domain: nodered.my.domain
base_url: http://forge.nodered.my.domain
api_url: http://forge:3000

#################################################
# Database Configuration                        #
#################################################

db:
  logging: false
  ## The database type: sqlite|postgres
  type: postgres
  ### SQLite options

  ## The database filename. Relative to $FLOWFORGE_HOME/var/
  ## Set to ':memory:' for a database that is wiped on restart
  # storage: forge.db

  ### Postgres options

  host: postgres
  user: forge
  password: secret

#################################################
# Project Driver Configuration                  #
#################################################

driver:
  ## The type of backend driver to use
  ##  Can be: localfs/docker/stub/k8s
  type: docker

  options:
    ## Options to be passed to the driver

    ### LocalFS options

    ## The first port number to assign to projects
    # start_port: 7880

    ## Path to find node executable if not on the default path
    # node_path: /usr/bin/node

    ### Docker options

    ## Docker socket path
    socket: /tmp/docker.sock
    # registry: hub.flowforge.com

    ### K8S options

    ## Kubectl conf file to contact the cluster
    # config_file: /opt/share/projects/flowforge/test/config
    # regsitry: hub.flowforge.com

#################################################
# Email Configuration                           #
#################################################

email:
  enabled: false
  debug: false
  smtp:
    host: localhost
    port: 587
    secure: false
    #auth:
    #    user: username
    #    pass: password

broker:
  url: mqtt://flowforge-broker:1883
  public_url: wss://mqtt.nodered.my.domain

fileStore:
  url: http://file-server:3001
```

The `data/settings.js` file inside the Node-RED container:
```


module.exports = {
    flowFile: 'flows.json',
    flowFilePretty: true,
    adminAuth: require('@flowforge/nr-launcher/adminAuth')({
        baseURL: 'http://test.nodered.my.domain',
        forgeURL: 'http://forge.nodered.my.domain',
        clientID: 'ffp_uDmOGs5AxnD6K4OqmifUVxgyQYcs9BUdH-yGa9b6uNQ',
        clientSecret: '-bMiIZbdhECazsyK12QTaNnKjC9HFdN-CoQvgvPiNzuijIctbS72Fa2PV5cZpH3k'
    }),
    credentialSecret: 'ade709596afb77728778156ea14d8e4cef0a99cfe100c96df4ca773094a51d2e',
    httpAdminRoot: '',
    ui: { path: '/ui'},
    disableEditor: false,
    
    
    tcpInAllowInboundConnections: false,
    udpInAllowInboundConnections: false,
    httpServerOptions: {
        "trust proxy": true
    },
    storageModule: require('@flowforge/nr-launcher/storage'),
    httpStorage: {
        projectID: '2bd680ff-719f-45b8-9956-80cf379761fd',
        baseURL: 'http://forge:3000/storage',
        token: 'fft_CO7P_-DkLCv5EVnwJOPtqZprT6-8s1tVEzxrmsRu9OU'
    },
    
    logging: {
        console: { level: 'info', metric: false, audit: false, handler: () => {
                const levelNames = {
                    10: "fatal",
                    20: "error",
                    30: "warn",
                    40: "info",
                    50: "debug",
                    60: "trace",
                    98: "audit",
                    99: "metric"
                }
                return (msg) => {
                    let message = msg.msg;
                    try {
                        if (typeof message === 'object' && message !== null && message.toString() === '[object Object]' && message.message) {
                            message = message.message;
                        }
                    } catch(e){
                        message = 'Exception trying to log: '+util.inspect(message);
                    }
                    console.log(JSON.stringify({
                        ts: Date.now(),
                        level: levelNames[msg.level],
                        type: msg.type,
                        name: msg.name,
                        id:msg.id,
                        msg: message
                    }))
                }
            }

        },
        auditLogger: {
            level: 'off', audit: true, handler: require('@flowforge/nr-launcher/auditLogger'),
            loggingURL: 'http://forge:3000/logging',
            projectID: '2bd680ff-719f-45b8-9956-80cf379761fd',
            token: 'fft_CO7P_-DkLCv5EVnwJOPtqZprT6-8s1tVEzxrmsRu9OU'
        }
    },
    nodesDir: ["/usr/src/flowforge-nr-launcher/node_modules/@flowforge/nr-launcher/node_modules/@flowforge/nr-project-nodes","/usr/src/flowforge-nr-launcher/node_modules/@flowforge/nr-project-nodes","/usr/src/flowforge-nr-launcher/node_modules/@flowforge/nr-launcher/node_modules/@flowforge/nr-file-nodes","/usr/src/flowforge-nr-launcher/node_modules/@flowforge/nr-file-nodes","/usr/src/flowforge-nr-launcher/node_modules/@flowforge/nr-launcher"],
    "forge-light": {"launcherVersion":"1.5.0","forgeURL":"http://forge.nodered.my.domain","projectURL":"http://forge.nodered.my.domain/project/2bd680ff-719f-45b8-9956-80cf379761fd"},
    editorTheme: {
        theme: 'forge-light',
        page: {
            title: 'FlowForge',
            favicon: "/theme/css/favicon.ico",
        },
        header: {
            title: 'test',
            url: 'http://forge.nodered.my.domain/project/2bd680ff-719f-45b8-9956-80cf379761fd'
        },
        logout: {
            redirect: 'http://forge.nodered.my.domain/project/2bd680ff-719f-45b8-9956-80cf379761fd'
        },
        codeEditor: {
            lib: 'monaco'
        },
        
        tours: true
    },
    nodesExcludes: [],
    externalModules: {
        // autoInstall: true,
        palette: {
            allowInstall: true,
            allowUpload: false,
            denyList: [],
            allowList: ["*"]
        },
        modules: {
            allowInstall: true,
            denyList: [],
            allowList: ["*"]
        }
    },
    functionExternalModules: true,
    flowforge: {
        forgeURL: 'http://forge.nodered.my.domain',
        teamID:  'Y3B28p2Nae',
        projectID: '2bd680ff-719f-45b8-9956-80cf379761fd',
        launcherVersion: '1.5.0',
        fileStore: {"url":"http://file-server:3001","token":"fft_CO7P_-DkLCv5EVnwJOPtqZprT6-8s1tVEzxrmsRu9OU"},
        
    },
    runtimeState: {
        enabled: true,
        ui: true
    }
}
```

**Comment by hardillb** [comment]
Hello, thanks for raising this.

Can I just check, the `flowforge.yml` file you've provided shows

```
port: 3000
host: 0.0.0.0
domain: nodered.my.domain
base_url: http://forge.nodered.my.domain
api_url: http://forge:3000
...
```

Did you remember to change the `base_url` value to start with `https://`? The `/data/settings.js` also still has `http://` which implies not.

If you fix the `flowforge.yml` you will need to delete the initial test project (as some of the URL values are stored in the database at creation time and not updated if you change the config) and create a new one after restarting the forge app container to pick up the changes.

**Comment by dvdjaco** [comment]
Indeed, that worked! I originally tried setting base_url to https but it wasn't working due to a different error, so I switched it back to the documented configuration (I was starting the stack with `docker compose up -d`, without the `-p flowforge` option, which caused a name resolution error from the nodered container). Anyway, I see that the documentation is already fixed in [24268aa](https://github.com/flowforge/flowforge/commit/24268aa5b7c02aa7e811c576b0ca68c19b323154), great!

Thanks for the quick reply!
